### PR TITLE
VideoPlayer.tsx: Don't newline subtitles. (Help yomitan get the entire subtitle)

### PR DESCRIPTION
**Pull request by Viterkim**
Related to the issue i had, after still struggling and not being able to do anything with yomitan, i decided to just fix it here.
https://github.com/killergerbah/asbplayer/issues/459#issuecomment-2242794235

image
image

Works!

### CODE REVIEW

**Comment by Viterkim** [comment]
Also this has been driving me crazy, trying to run it locally.
image
The extension doesn't seem to work/thinks its a different site even though I changed this. 

image

the streaming video options also dissapears.

There's also hardcoded things, preventing you from simply running it on localhost, it has to be at localhost/asbplayer
```
    GET http://localhost/asbplayer/static/js/main.62a96a7c.js net::ERR_ABORTED 404 (Not Found)
script.js:1 Ignoring Event: localhost
```
 
Was thinking it would be convenient to run it/test everything + run it locally if internet goes down :)

**Comment by killergerbah** [comment]
Have you tried fixing this problem using the regex-based text replacement feature instead?
Under misc settings you can configure asbplayer to replace all `\n` characters with a space ` `.

About the extension and locally-hosted asbplayer website - prod versions of the extension (like from Chrome web store) do not recognize localhost due to restrictions on the webstore. You can build a dev version of the extension instead if you want it to work with a locally-hosted asbplayer website.

**Comment by Viterkim** [comment]
That makes it into 1 long line yea, but that's definitely worse to read. (I think basically all long subtitles are stacked). So not ideal :/
image


And i guess for making it work offline, it would be way easier to just add a serviceworker and add "progressive web app" support, that way the browser would cache the app even if you don't have internet.

**Comment by killergerbah** [comment]
Have you tried this setting on Yomitan?
image

I am hesitant to make changes just to workaround specific problems with other software.

**Comment by Viterkim** [comment]
Yes i tried that, it adds random text from other sentences to the line.


I'm not really sure what the bad thing about adding it would be :D
I have another friend who has the same issue but who doesn't use github. I think it's valid to be able to get the full sentence without using the extension? (And i think for most people, the only reason why they're using a browser media player, is for pop up dictonaries).

**Comment by killergerbah** [comment]
Okay I see your point.
I'll come back to this - I want to be reasonably sure this change is safe to make.

**Comment by Viterkim** [comment]
> Okay I see your point. I'll come back to this - I want to be reasonably sure this change is safe to make.

Sure! Also don't want to pressure anything! 
Maybe make it an option in the misc settings to be 100% sure? (So maybe default on, then `Don't turn newlines into tags in subtitles` or something

File: common/app/components/VideoPlayer.tsx [hunk 1]
```diff
@@ -164,7 +164,9 @@ const showingSubtitleHtml = (
 `;
     }
 
-    return `<span style="${subtitleStyles}" class="${subtitleClasses}">${subtitle.text}</span>`;
+    const lines = subtitle.text.split('\n');
```
Comment (killergerbah) [inline common/app/components/VideoPlayer.tsx:168]: This change should be repeated on line 198 to get the same behavior when DOM caching is disabled.

**Review by killergerbah — CHANGES_REQUESTED** [review]

File: common/app/components/VideoPlayer.tsx (hunk 1 above)
Comment (Viterkim) [reply common/app/components/VideoPlayer.tsx:168]: image
image

**Review by killergerbah — CHANGES_REQUESTED** [review]
It looks like there's a bug: multiple subtitle tracks are rendered with extra space between them now.

**Comment by Viterkim** [comment]
> It looks like there's a bug: multiple subtitle tracks are rendered with extra space between them now.

Do you have a subtitle file i could try out and see?

**Comment by Viterkim** [comment]
Here's my branch, and master. (With & without the dom option). (No difference)
image
image
image
image

**Comment by Viterkim** [comment]
Or is it this? foun some subtitle online with dual, but it seems really weird
image
seems to only happen on my branch, with `Pre-cache Subtitle DOM` on
image

I'll try to fix it

**Comment by Viterkim** [comment]
image
```
    line-height: normal;
    display: inline;
```
seems to do it

**Comment by Viterkim** [comment]
Actually seems like you can get away with just setting
```
.subtitle-line {
    display: inline;
}
```

I don't know what's best :)

**Comment by Viterkim** [comment]
image
ok should work now

**Comment by Viterkim** [comment]
image

**Comment by killergerbah** [comment]
Can you run Prettier on VideoPlayer.tsx? I can merge after that.

**Comment by killergerbah** [comment]
Thanks!

**Pull request merged**